### PR TITLE
create toDo main part

### PR DESCRIPTION
**Pull request by NikitaVysochin**

### CODE REVIEW

File: ind.css [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+ul[class],
+ol[class],
+li,
+figure,
+figcaption,
+blockquote,
+dl,
+dd {
+  margin: 0;
+}
```
Comment (usatovaDasha) [inline ind.css:27]: add empty lines between blocks

File: ind.css [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+  box-shadow: 0 0 10px rgba(233, 233, 233, 0.5);
+}
+
+.AllDoList {
+  justify-content: space-around;
+  display: flex;
+  width: 80%;
+  margin: auto auto;
+}
+.list1 {
```
Comment (usatovaDasha) [inline ind.css]: correct name for classes

File: test-module-2.js [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+let but = document.querySelector(".main-but");
+let empty = document.querySelector(".empty");
+let empty2 = document.querySelector(".empty2");
+let arr = JSON.parse(localStorage.getItem("arr")) || [];
+let InpVal = "";
+let input = null;
+let editValue = "";
+let proindex;
+
+window.onload = function init() {
```
Comment (usatovaDasha) [inline test-module-2.js]: es6

File: test-module-2.js [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+  input = document.querySelector(".main-inp");
+  input.addEventListener("change", upDatevalue);
+  DoIt();
+};
+
+but.addEventListener("click", onClickMainBut);
+
+function onClickMainBut() {
```
Comment (usatovaDasha) [inline test-module-2.js]: es6

File: test-module-2.js [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+  input.value = "";
+  DoIt();
+  localStorage.setItem("arr", JSON.stringify(arr));
+}
+
+upDatevalue = (event) => {
+  InpVal = event.target.value;
+};
+
+function DoIt() {
```
Comment (usatovaDasha) [inline test-module-2.js]: es6 all functions

File: test-module-2.js [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+  while (list1.firstChild) {
+    list1.removeChild(list1.firstChild);
+  }
+  arr.map((item, index) => {
+    let newMain = document.createElement("div");
```
Comment (usatovaDasha) [inline test-module-2.js]: why let?

File: test-module-2.js [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+    newMain.className = "main-str";
+    newMain.id = `main-str-${index}`;
+    list1.appendChild(newMain);
+
+    let str = document.createElement("div");
```
Comment (usatovaDasha) [inline test-module-2.js]: why let?

File: test-module-2.js [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+    str.id = `str-${index}`;
+    if (arr[index].isCheck) {
```
Comment (usatovaDasha) [inline test-module-2.js]: use ternary

File: test-module-2.js [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+      str.className = "str2";
+    } else {
+      str.className = "str";
+    }
+
+    str.innerHTML = item.text;
```
Comment (usatovaDasha) [inline test-module-2.js]: use destructurization item

File: test-module-2.js [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+    newMain.appendChild(str);
+
+    strButRed = document.createElement("div");
+    strButRed.className = "str-but-red";
+    let img = document.createElement("img");
```
Comment (usatovaDasha) [inline test-module-2.js]: why let?

File: test-module-2.js [hunk 11]
```diff
@@ -0,0 +1,160 @@
...
+    img.src = "img/listing_to_do_list_document_file_icon_147456.png";
+    strButRed.appendChild(img);
+    strButRed.addEventListener("click", () => redact(str, index));
+    newMain.appendChild(strButRed);
+    if (arr[index].isCheck) {
+      strButRed.style = "display:none";
+    }
+
+    let strButDel = document.createElement("div");
```
Comment (usatovaDasha) [inline test-module-2.js]: why let?

File: test-module-2.js [hunk 12]
```diff
@@ -0,0 +1,160 @@
...
+    strButDel.className = "str-but-del";
+    let img3 = document.createElement("img");
```
Comment (usatovaDasha) [inline test-module-2.js]: why let?

Comment (usatovaDasha) [inline test-module-2.js]: correct name variables

File: test-module-2.js [hunk 13]
```diff
@@ -0,0 +1,160 @@
...
+    img3.src = "img/fe.png";
+    strButDel.appendChild(img3);
+    strButDel.addEventListener("click", () => deleteStr(newMain, index));
+    newMain.appendChild(strButDel);
+
+    let strButDid = document.createElement("div");
```
Comment (usatovaDasha) [inline test-module-2.js]: why let?

File: test-module-2.js [hunk 14]
```diff
@@ -0,0 +1,160 @@
...
+    strButDid.className = "str-but-did";
+    let img2 = document.createElement("img");
```
Comment (usatovaDasha) [inline test-module-2.js]: why let?

File: test-module-2.js [hunk 15]
```diff
@@ -0,0 +1,160 @@
...
+    img2.src = "img/to_do_icon_153795.png";
+    strButDid.appendChild(img2);
+    strButDid.checked = item.isCheck;
+    strButDid.addEventListener("click", () => checkbox(index));
+    newMain.appendChild(strButDid);
+
+    arr.sort(function (a, b) {
```
Comment (usatovaDasha) [inline test-module-2.js]: es6

File: test-module-2.js [hunk 16]
```diff
@@ -0,0 +1,160 @@
...
+      return a.isCheck - b.isCheck;
+    });
+    empty2.addEventListener("click", () => dobav(index, str));
+    localStorage.setItem("arr", JSON.stringify(arr));
+  });
+}
+
+function checkbox(index) {
+  arr[index].isCheck = !arr[index].isCheck;
+  arr.sort(function (a, b) {
```
Comment (usatovaDasha) [inline test-module-2.js]: why need this sort?

File: test-module-2.js [hunk 17]
```diff
@@ -0,0 +1,160 @@
...
+}
+
+function redact(str, index) {
+  let newInp = document.createElement("input");
+  newInp.className = "inp2";
+  newInp.value = arr[index].text;
+  str.innerHTML = "";
+  str.appendChild(newInp);
+
+  newInp.addEventListener("blur", function () {
```
Comment (usatovaDasha) [inline test-module-2.js]: es6

File: test-module-2.js (hunk 16 above)
Comment (NikitaVysochin) [reply test-module-2.js]: exclude bug with sort checked elements

File: test-module-2.js [hunk 18]
```diff
@@ -51,11 +51,7 @@ const DoIt = () => {
 
     const str = document.createElement("div");
     str.id = `str-${index}`;
-    if (arr[index].isCheck) {
-      str.className = "str2";
-    } else {
-      str.className = "str";
-    }
+    arr[index].isCheck ? str.className = "str2" : str.className = "str";
```
Comment (usatovaDasha) [inline test-module-2.js]: str.className = arr[index].isCheck ? "str2" : "str";

File: test-module-2.js (hunk 9 above)
Comment (usatovaDasha) [reply test-module-2.js]: !!!!

File: test-module-2.js [hunk 19]
```diff
@@ -44,16 +44,17 @@ const DoIt = () => {
     mainList.removeChild(mainList.firstChild);
   }
   arr.map((item, index) => {
+    const {text, isCheck} = item;
```
Comment (usatovaDasha) [inline test-module-2.js]: spaces

File: test-module-2.js [hunk 20]
```diff
@@ -44,16 +44,17 @@ const DoIt = () => {
...
     newMain.className = "main-str";
     newMain.id = `main-str-${index}`;
     mainList.appendChild(newMain);
 
     const str = document.createElement("div");
     str.id = `str-${index}`;
-    arr[index].isCheck ? str.className = "str2" : str.className = "str";
-
-    str.innerHTML = item.text;
+    isCheck ? str.className = "str2" : str.className = "str";
```
Comment (usatovaDasha) [inline test-module-2.js]: i wrote what you need fix

**Pull request merged**